analyses/mwi/dryspells_trigger/15_mwi_obs_trigger_monitoring.md

```diff
@@ -0,0 +1,154 @@
+```python
+%load_ext autoreload
+%autoreload 2
+
+import os
+from pathlib import Path
+import sys
+from datetime import date
+
+path_mod = f"{Path(os.path.dirname(os.path.abspath(''))).parents[1]}/"
+sys.path.append(path_mod)
+from src.indicators.drought.arc2_precipitation import DrySpells
+
+import geopandas as gpd
+import xarray as xr
+import numpy as np
+# for plotting
+import matplotlib.pyplot as plt
+from matplotlib import colors
+```
+
+The first we will do is setup the 3 monitors, which are going to use the 3 separate types of aggregation. We only need to download the data once for each because they all depend on the same raw data from ARC2. Currently, I've kept them using the 2018 monitoring period as an example because we have not reached the period we want to monitor. To monitor in 2022, all we have to do is set `monitoring_start = "2021-01-07"` and `monitoring_end = date.today()`.
+
+```python
+## Global variables for all monitoring
+
+POLY_PATH = os.path.join(
+    os.getenv('AA_DATA_DIR'),
+    'public',
+    'processed',
+    'mwi',
+    'cod_ab',
+    'mwi_drought_adm2.gpkg'
+)
+
+# Admin3 file for mapping
+gdf_adm3 = gpd.read_file(os.path.join(
+    os.getenv('AA_DATA_DIR'),
+    'public',
+    'raw',
+    'mwi',
+    'cod_ab',
+    'mwi_adm_nso_20181016_shp',
+    'mwi_admbnda_adm3_nso_20181016.shp'
+))
+
+gdf_adm3 = gdf_adm3[gdf_adm3.ADM1_PCODE == "MW3"]
+
+MONITORING_START = "2018-01-07"
+MONITORING_END = "2018-03-07"
+RANGE_X = ("32E", "36E")
+RANGE_Y = ("20S", "5S")
+
+# centroid method
+arc2_centr = DrySpells(
+    country_iso3 = "mwi",
+    monitoring_start = MONITORING_START,
+    monitoring_end = MONITORING_END,
+    range_x = RANGE_X,
+    range_y = RANGE_Y
+)
+
+arc2_centr.download_data()
+
+# touching method
+arc2_touch = DrySpells(
+    country_iso3 = "mwi",
+    monitoring_start = MONITORING_START,
+    monitoring_end = MONITORING_END,
+    range_x = RANGE_X,
+    range_y = RANGE_Y,
+    agg_method = "touching"
+)
+
+# approximate mask
+arc2_approx = DrySpells(
+    country_iso3 = "mwi",
+    monitoring_start = MONITORING_START,
+    monitoring_end = MONITORING_END,
+    range_x = RANGE_X,
+    range_y = RANGE_Y,
+    agg_method = "approximate_mask"
+)
+```
+
+Now with each of these, we can just re-process our data and calculate rolling sums and dry spells.
+
+```python
+arc2_centr.aggregate_data(POLY_PATH, "ADM2_PCODE")
+arc2_centr.calculate_rolling_sum()
```

Comment:
I was just trying to run the notebook and got an error here: `first argument must be an iterable of pandas objects, you passed an object of type "DataFrame"` 
According to [SO](https://stackoverflow.com/questions/39534676/typeerror-first-argument-must-be-an-iterable-of-pandas-objects-you-passed-an-o/50459002) had to make a list object when concatenating in `arc2_precipitation.py` in line 582 and 594. That did solve the problem for me

Reply:
In other work got rid of that code, so no worries now as it all runs ha!